reclaimer/sounds/Encoder.py

```diff
@@ -0,0 +1,221 @@
+
+                                                                        #While there are still blocks to process
+        while vorbis_analysis_blockout(self.vorbisencoderstate, self.vorbisblock) == 1:
+            vorbis_analysis(self.vorbisblock, None)                     #Analyze the block
+            vorbis_bitrate_addblock(self.vorbisblock)                   #Push the block
+                                                                        #while there are packets to process
+            while vorbis_bitrate_flushpacket(self.vorbisencoderstate, self.packet):
+                ogg_stream_packetin(self.streamstate, self.packet)          #Get a packet
+                while not eos:                                              #While there is more to stream
+                    result = ogg_stream_pageout(self.streamstate, self.page)    #Get a page
+                    if result == 0: break                                           #if there are no pages, get next packet
+                    self.Output += self.page.header[0:self.page.header_len] #Write output
+                    self.Output += self.page.body[0:self.page.body_len]     #Write Output
+                    if ogg_page_eos(self.page): eos = True                  #If at end of stream, quit looping
+
+    #Quality of life functions:
+
+    def AddVorbisComment(self, key, value = ""):                    #Add a comment to the stream (Bugged)
+        if self.log: print("Adding Vorbis Comment: " + key + " " + value)
+        if value == "":                                                 #if there was only one arg
+            libvorbis.vorbis_comment_add(self.vorbiscomment, key)           #add the comment
+        else:                                                           #If tag and comment ("key", "value")
+            libvorbis.vorbis_comment_add_tag(self.vorbiscomment, key, value)#add the comment
+
+    def ToByteArray(self):                                          #Convert to byte array
+        return bytearray(self.Output)
+
+
+    #Error "Handling" functions.
+    #ToDo : actually handle the error instead of just print.
+
+    def _GENERIC(self):
+        print("An unknown error has occured.")
+        self.__exit__()
+
+    def _OGG_STREAM_FAULT(self):
+        print("Unable to initialize the ogg stream.")
+        self.__exit__()
+
+    def _OGG_ANALYSIS_HEAD_OUT_FAULT(self):
+        print("OGG Header packet creation failed.")
+        self.__exit__()
+
+    def _OGG_STREAM_FAIL(self):
+        print("Unable to write packet to stream.")
+        self.__exit__()
+
+    def _V_BLOCK_INIT_FAULT(self):
+        print("Unable to initialize the vorbis block.")
+        self.__exit__()
+
+    def _V_ANALYSIS_FAULT(self):
+        print("Unable to initialize the encoders analysis state.")
+        self.__exit__()
+
+    def _V_ENCODE_INIT_FAULT(self):
+        print("Internal logic Fault. Likely invalid quality request. (-0.1 - 1.0)")
+        self.__exit__()
```

Comment:
Could we instead of using these throw exceptions? Don't be worried about creating some new exception types. The thing that calls and uses this class should handle these exceptions itself anyway.